pkgs/development/python-modules/torpy/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, cryptography
+, pytest
+, requests
+}:
+
+buildPythonPackage rec {
+  pname = "torpy";
+  version = "1.1.6";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "torpyorg";
+    repo = "torpy";
+    rev = "v${version}";
+    sha256 = "sha256-Ni7GcpkxzAMtP4wBOFsi4KnxK+nC0XCZR/2Z/eS/C+w=";
+  };
+
+  propagatedBuildInputs = [
+    cryptography
+    pytest
```

Comment:
```suggestion
```

Reply:
When I do the suggested change I get a build error

```
tests.unittest.test_hiddenservice (unittest.loader._FailedTest) ... ERROR
tests.unittest.test_parsers (unittest.loader._FailedTest) ... ERROR

======================================================================
ERROR: test_consensus (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_consensus
Traceback (most recent call last):
  File "/nix/store/h708i6480kns43mmky00vkiqdnc1lyhk-python3-3.9.13/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/private/tmp/nix-build-python3.9-torpy-v1.1.6.drv-0/source/tests/unittest/test_consensus.py", line 1, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'
```

Reply:
pytest is already in checkInputs which only get added to the environment when running the tests. I think the package is triggering the tests while being build already. Can you double check that? Worst case we would need to move pytest to buildInputs because it shoudn't be added to propagated to make overrides for it easier.

Reply:
Yes, the package is running the tests when it is being built.
[buildlog.txt](https://github.com/NixOS/nixpkgs/files/9176330/buildlog.txt)

Reply:
But the tests are not run when I'm installing without nix, with 

    python3 -m venv torpyenv
    source ./torpyenv/bin/activate
    pip install --upgrade pip
    pip install cryptography
    git clone https://github.com/torpyorg/torpy
    cd torpy
    python setup.py install


I have to do `python setup.py test` to run the tests

Reply:
``python setup.py test`` should only be done in checkPhase where pytest is available.

Reply:
I pushed the suggested change (I removed pytest from propagatedBuildInputs).